src/CppAst/CppModelBuilder.cs

```diff
@@ -1975,41 +2065,87 @@ protected void WarningUnhandled(CXCursor cursor, CXCursor parent)
             RootCompilation.Diagnostics.Warning($"Unhandled declaration: {cursor.Kind}/{cursor} in {parent}.", cppLocation);
         }
 
-        private List<CppType> ParseTemplateSpecializedArguments(CXCursor cursor, CXType type, CXClientData data)
+        private (List<CppType>, List<CppTemplateArgument>) ParseTemplateSpecializedArguments(CXCursor cursor, CXType type, CXClientData data)
```

Comment:
I am a little bit confused in terms of what the updated function is doing.  `ParseTemplateSpecializedArguments()` is supposed to return specialized arguments instead of template parameters. Why is the return value split between typed and non-typed arguments?

Reply:
I'll rename the function as it should parse both specialized and generic parameters